### ISSUE TITLE
Support for angular elements in series name and yAxis.title.text

### ISSUE THREAD

**Issue by fullstackdeveloperbilbao** [highcharts/highcharts-angular#376]
### Describe the bug
I am trying to add angular elements as DOM elements to serie.name and yAxis.title.text. 
serie.name accepts html by default and it parses and shows correctly, however when passing an angular element nothing is shown.
yAxis.title accepts useHtml property, which I have set to true.  When settings text to the angular element I am receiving next error: 

Highcharts warning #33: www.highcharts.com/errors/33/?Invalid%20tagName%20in%20config=DEMO-ELEMENT
 - Invalid tagName in config: DEMO-ELEMENT

### Expected behavior
I expect to have some method that behind the scenes uses appendChild(demoElement) and shows it.
With normal js you can use for example body.appendChild(demoElement) and the element will work as a normal component but it has been added dynamically. This would allow using pipes... inside serie.name and yAxis.title.text.

### Demo
1- Create the component that will be used by angular elements:

```
// demo.component.ts
@Component({
  selector: 'app-demo',
  templateUrl: './demo.component.html',
})
export class DemoComponent implements OnInit {
  counter = 0;

 ngOnInit() {
  setInterval(() => {
  this.counter++;
  }
  ,1000);
}

```
```
<!--  demo.component.html -->

<div>{{counter}}</div>
```
```
// parent.component.ts

...
import {
  createCustomElement,
  NgElement,
  WithProperties,
} from '@angular/elements';
...
@Component({
  selector: 'app-parent',
  templateUrl: './parent.component.html',
})
export class ParentComponent implements OnInit {
 private injector = inject(Injector);

 constructor() {
  this.defineDemoElement();
 }

 ngOnInit() {
 const demoEl: NgElement & WithProperties<DemoComponent> =
      document.createElement('demo-element') as any;
  this.chartOptions = {
          ....
          yAxis: [
           {
            title: {
              text: demoEl,
              useHTML: true,
              ...
          },
         ]
     }
  }
 }

 private defineDemoElement() {
    if (isNil(customElements.get('demo-element'))) {
      const demoElement = createCustomElement(DemoComponent, {
        injector: this.injector,
      });
      customElements.define('demo-element', demoElement);
    }
  }
}
```

#### Steps required to recreate the problem in the demo:
The error is shown when the chart loads.

### Setup used
 - NodeJS version: 18.13.0
 - Angular Elements version: ^16.2.12
 - Angular version: ^16.2.0
 - TypeScript version: ~5.1.3

**Comment by fullstackdeveloperbilbao** [comment]
[Here](https://github.com/fullstackdeveloperbilbao/angular-elements-highcharts.git) there is a working example. You just need to:
1- Run npm i
2- Run npm start

**Comment by fullstackdeveloperbilbao** [comment]
I have found the solution. 

To use custom elements just need to add to main.ts:

```
if (Highcharts && Highcharts.AST) {
  Highcharts.AST.allowedTags = [...Highcharts.AST.allowedTags, "demo-element"];
}
```

Then you can use on text property next to useHtml: true the outerHtml of your element:

```
yAxis: [
        {
          labels: {
            format: "{value}",
          },
          title: {
            text: demoEl.outerHTML,
            useHTML: true,
          },
        },
      ],
```
For series.name it is needed to make an additional change mto the legend property:

```
series: [
        {
          name: demoEl.outerHTML,
          showInLegend: true,
          type: "spline",
          data: data,
          marker: {
            enabled: false,
          },
        },
      ],
```
 ```
legend: {
        useHTML: true,
      },
```
Working solution is in the [repo](https://github.com/fullstackdeveloperbilbao/angular-elements-highcharts.git).

It would make me happy if I could contribute to highcharts documentation as it would be my first contribution to an official repo. If you agree I can prepare some documentation about angular elements and dynamic components added to highcharts as this allows using translation pipes and other angular functionallities to highcharts.

**Comment by fullstackdeveloperbilbao** [comment]
This issue solution is the solution for #371 .There js listens to lang changes and updates the chartOptions accordingly. The solution is to create a component that accepts the translation key and use it in the lang properties. I have pushed to the [demo repo](https://github.com/fullstackdeveloperbilbao/angular-elements-highcharts) an example with downloadPNG property.

Also is the solution to #232

**Comment by karolkolodziej** [comment]
Hi @fullstackdeveloperbilbao apologies for my late reply and thank you for describing everything here! 
if you want to contribute and describe everything in the docs feel free to create a PR with it!

**Comment by fullstackdeveloperbilbao** [comment]
Hi @karolkolodziej , the docs have been updated in the PR #377. If I need to modify something please let me know.
Thx.

**Comment by karolkolodziej** [comment]
closed by #377
